File: backend/src/modules/XiHan.BasicApp.AI/Domain/Enums/KnowledgeEnums.cs
// Copyright (c) 2021-Present XiHanFun and contributors.
// Licensed under the MIT License. See LICENSE in the project root for license information.

using System.ComponentModel;

namespace XiHan.BasicApp.AI.Domain.Enums;

/// <summary>
/// 知识文档来源类型
/// </summary>
public enum KnowledgeSourceType
{
    /// <summary>
    /// 粘贴文本
    /// </summary>
    [Description("粘贴文本")]
    PasteText = 0,

    /// <summary>
    /// 上传文件（前端读取为文本后提交）
    /// </summary>
    [Description("上传文件")]
    UploadFile = 1
}

/// <summary>
/// 知识文档索引状态
/// </summary>
public enum KnowledgeIndexStatus
{
    /// <summary>
    /// 待索引
    /// </summary>
    [Description("待索引")]
    Pending = 0,

    /// <summary>
    /// 已索引
    /// </summary>
    [Description("已索引")]
    Indexed = 1,

    /// <summary>
    /// 索引失败
    /// </summary>
    [Description("失败")]
    Failed = 2
}

File: frontend/src/modules/ai/api/knowledge.enums.ts
/**
 * 知识库枚举（与后端全局 JsonStringEnumConverter 序列化值一致，成员名即 wire value）。
 */

/** 知识文档来源类型 */
export enum KnowledgeSourceType {
  PasteText = 'PasteText',
  UploadFile = 'UploadFile',
}

/** 知识文档索引状态 */
export enum KnowledgeIndexStatus {
  Pending = 'Pending',
  Indexed = 'Indexed',
  Failed = 'Failed',
}

/** 来源类型选项 */
export const KNOWLEDGE_SOURCE_TYPE_OPTIONS = [
  { label: '粘贴文本', value: KnowledgeSourceType.PasteText },
  { label: '上传文件', value: KnowledgeSourceType.UploadFile },
]

/** 索引状态选项 */
export const KNOWLEDGE_INDEX_STATUS_OPTIONS = [
  { label: '待索引', value: KnowledgeIndexStatus.Pending },
  { label: '已索引', value: KnowledgeIndexStatus.Indexed },
  { label: '失败', value: KnowledgeIndexStatus.Failed },
]
